lib/Runtime/Types/JavascriptStaticEnumerator.cpp

```diff
@@ -0,0 +1,90 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeTypePch.h"
+#include "Library/JavascriptSymbol.h"
+
+namespace Js
+{
+
+    bool JavascriptStaticEnumerator::Initialize(JavascriptEnumerator * prefixEnumerator, ArrayObject * arrayToEnumerate, 
+        DynamicObject * objectToEnumerate, EnumeratorFlags flags, ScriptContext * requestContext)
```

Comment:
Can we in debug build, save the requestContext, and assert the prefixEnumerator returned propertyString doesn't need marshalling?